src/helpers/target-scope.ts

```diff
@@ -2,33 +2,43 @@ import path from "path";
 import { Context } from "../types";
 import { Scope } from "../types/github";
 import { Target } from "../types/target";
+import { checkUserRepoPermissions } from "./user-permission";
 
-export function targetBuilder(context: Context, scope: Scope): Record<string, Target> {
+export async function targetBuilder(context: Context, scope: Scope): Promise<Record<string, Target>> {
   const { payload, config, logger } = context;
   const targetMap: Record<string, Target> = {};
 
-  const baseTargets: Target[] = config.defaultTargets.map((target) => {
+  const baseTargets: Target[] = [];
+  for (const target of config.defaultTargets) {
     const match = RegExp(/github\.com\/([^/]+)\/([^/]+)(\.git)?$/).exec(target.name);
     if (!match) {
       throw logger.error(`Invalid GitHub URL: ${target.name}`);
     }
     const owner = match[1];
     const repo = match[2].replace(".git", "");
-    return {
+
+    // Check Access to the base targets
+    const hasPermission = await checkUserRepoPermissions(context, owner, repo);
+
+    baseTargets.push({
       type: target.type || "main",
       owner,
       repo,
       localDir: path.join(owner, repo),
       url: target.name,
+      scope: target.scope === "REPO" ? Scope.REPO : Scope.ORG,
```

Comment:
This should definitely be implicit. Just check if repo config exists, if not, then org. The user doesn't need to specify this.

Reply:
If the user specifies /config <Editor_Instruction>, in case 1, if both org and repo configs are available, a PR will be created for both. In case 2, if only the org or repo config is available, a PR will be created for the available one. Is that correct?